### PR TITLE
Batch produce

### PR DESCRIPTION
**Pull request by simlay**
Initial stuff for batch produce to show difficulty. Depends on infinyon/fluvio#896, subtask of infinyon/fluvio#847.

### CODE REVIEW

File: fluvio/__init__.py
Comment (simlay) [inline fluvio/__init__.py:174]: You might need to [`typing.Tuple`](https://docs.python.org/3/library/typing.html#typing.Tuple). The python docs don't say when `builtin.tuple` was added but my guess is that the Python 3.8 CI is failing due to this.

**Review by simlay — COMMENTED** [review]
What I can't approve my "own" pr? @nicholastmosher Could you bump the version in `setup.py`? We can publish it once this is merged that way.

File: fluvio/__init__.py
Comment (nicholastmosher) [reply fluvio/__init__.py:174]: I actually originally tried that but when I use `typing.tuple` the build fails for me

Comment (nicholastmosher) [reply fluvio/__init__.py:174]: Oh let me try capital-T Tuple

Comment (nicholastmosher) [reply fluvio/__init__.py:174]: Ok that worked 👍

Comment (simlay) [reply fluvio/__init__.py]: Yeah, the type annotations in python are a bit bleh. I only like them for the documentation generation.

**Review by nicholastmosher — APPROVED** [review]
How about I "approve" it and I'll let you decide when to merge it? :P

**Pull request merged**
